### PR TITLE
hide embedded projects in the project list

### PR DESCRIPTION
**Pull request by laurelfan**

### CODE REVIEW

File: dashboard/app/views/levels/_project_ui.html.haml [hunk 1]
```diff
@@ -78,13 +78,17 @@
           });
           if (projects.length == 0) {
             createApp({projectTemplateLevelName: appOptions.level.projectTemplateLevelName,
-                       name: appOptions.level.projectTemplateLevelName});
+                       name: appOptions.level.projectTemplateLevelName,
+                       hidden: true});
           } else {
-            var app = projects[0];
+            currentApp = projects[0];
             // TODO this should just load the code in the workspace, not refresh
```
Comment (laurelfan) [inline dashboard/app/views/levels/_project_ui.html.haml]: I am going TODO this next

File: dashboard/app/views/projects/index.html.haml [hunk 2]
```diff
@@ -69,6 +69,10 @@
       $scope.projectsLoaded = true;
     }).catch($scope.genericError);
 
+    $scope.projectVisible = function(project) {
+      return (!project.hidden);
```
Comment (joshlory) [inline dashboard/app/views/projects/index.html.haml]: Filtering on the client side, not in the API?

Comment (sfilman) [inline dashboard/app/views/projects/index.html.haml]: This does seem like something that should be an API/service level change to
me.

Comment (laurelfan) [inline dashboard/app/views/projects/index.html.haml]: if the service doesn't return projects that are 'hidden' then we can't use those projects on the client side (ie. we can't load the code from the project, we can't save them, etc).

Note that "client" doesn't mean "user"... the client here is code that we are writing.

Comment (joshlory) [inline dashboard/app/views/projects/index.html.haml]: I could see an API where fetches for the hidden projects would still succeed, they just wouldn't be enumerated when querying.

Comment (laurelfan) [inline dashboard/app/views/projects/index.html.haml]: Ok... how do we know what to fetch? Do we need another api to store a list
of projects?

Comment (joshlory) [inline dashboard/app/views/projects/index.html.haml]: Either a `showHidden=true` param or maybe we really do need namespaces that can be used as filters in the .all() query.  This feels like scope creep so I'm fine with client-side for now, until we better understand what the API filters look like.

Comment (laurelfan) [inline dashboard/app/views/projects/index.html.haml]: Yeah, adding filter/search to the API may make sense, but I think if we do
it we will want it to be generic filter/search because from the API's point
of view the properties are not meaningful (so basically the client would
give the server a query), and I can see us wanting to filter on other
things (like projectTemplate name), not just hidden. We could also add
filtering to the js "service" (ie. the projectsService js object) first if
we think it's messy to do this in the view.
